lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4825,6 +4825,137 @@ Instruction *SPIRVToLLVM::transRelational(SPIRVInstruction *I, BasicBlock *BB) {
   return cast<Instruction>(Mutator.getMutated());
 }
 
+std::optional<SPIRVModuleReport> getSpirvReport(std::istream &IS) {
```

Comment:
This is a self-contained change. Is it possible to move this to a separate .cpp file (something like SPIRVAnalyzer.cpp)? Of course, we need to consider code bloating.

Thanks

Reply:
That's a good point in general, thank you. However, I'm afraid this is a much wider issue, because lib/SPIRV/SPIRVReader.cpp is bloated indeed. This feature is not exactly self-contained, rather it is integrated with existing internal API, so probably the question of splitting of lib/SPIRV/SPIRVReader.cpp deserves actually a separate PR that should address a question of proper management of features presented in lib/SPIRV/SPIRVReader.cpp. If we started discussing how to proper split lib/SPIRV/SPIRVReader.cpp as a part of this PR it would effectively block the PR instead of implementing this as a small and incremental change that uses existing internal api.

Reply:
I agree that this can be done as separate refactoring. A TODO can be handy here.